tensorzero-core/src/observability/mod.rs

```diff
@@ -133,6 +134,9 @@ struct CustomTracer {
 pub struct TracerWrapper {
     default_tracer: SdkTracer,
     default_provider: SdkTracerProvider,
+    // Static headers from the config that are always included (can be overridden by dynamic headers)
+    // Wrapped in Arc<RwLock> so we can update them after initialization (e.g. in the gateway after loading config)
+    static_otlp_traces_extra_headers: Arc<RwLock<MetadataMap>>,
```

Comment:
This can be an `Arc<OnceCell<MetadataMap>>`, which is cheaper to access. Inside of `build_with_context`, you can use `.get()`, and check if it's `Some`